_test/test_utilities_herbert/test_str_to_iarray.m

```diff
@@ -0,0 +1,187 @@
+            iarr = str_to_iarray ('-8--3');
+            assertEqual (iarr, -8:-3)
+        end
+        
+        function test_non_Matlab_range_neg2neg_descend (~)
+            iarr = str_to_iarray ('-3--8');
+            assertEqual (iarr, -3:-1:-8)
+        end
+       
+        function test_non_Matlab_range_neg2neg_single (~)
+            % Tough test of parsing?
+            iarr = str_to_iarray ('-3--3');
+            assertEqual (iarr, -3)
+        end
+       
+        function test_non_Matlab_range_neg2neg_single_FAIL (~)
+            % Tough test of parsing? Catch erroneous range indicator at front
+            func = @()str_to_iarray ('--3--3');
+            assertExceptionThrown (func, 'HERBERT:str_to_iarray:invalid_argument');
+        end
+       
+        
+        %--------------------------------------------------------------------------
+        % Tokens with whitespace
+        %--------------------------------------------------------------------------
+        function test_single_number_WS (~)
+            iarr = str_to_iarray ('  45  ');
+            assertEqual (iarr, 45)
+        end
+        
+        function test_range_with_positive_stride_rhs_gt_lhs_WS (~)
+            iarr = str_to_iarray ('  3:2:14  ');
+            assertEqual (iarr, 3:2:14)
+        end
+                
+        function test_range_with_positive_stride_rhs_gt_lhs_WSfail (~)
+            % Should fail as we do not permit whitespace within the token
+            func = @()str_to_iarray (' 3 :2: 14');
+            assertExceptionThrown (func, 'HERBERT:str_to_iarray:invalid_argument');
+        end
+        
+        function test_non_Matlab_range_with_positive_stride_rhs_gt_lhs_WS (~)
+            iarr = str_to_iarray ('  3-14  ');
+            assertEqual (iarr, 3:14)
+        end
+                
+        function test_non_Matlab_range_with_positive_stride_rhs_gt_lhs_WSfail (~)
+            % Should fail as we do not permit whitespace within the token
+            func = @()str_to_iarray (' 3 - 14');
+            assertExceptionThrown (func, 'HERBERT:str_to_iarray:invalid_argument');
+        end
+        
+        %--------------------------------------------------------------------------
+        % Multiple tokens
+        %--------------------------------------------------------------------------
+        function test_two_numbers (~)
+            iarr = str_to_iarray ('45,46');
+            assertEqual (iarr, [45,46])
+        end
+        
+        function test_mixed_format_ranges (~)
+            iarr = str_to_iarray ('14:-2:5, 19 -3--6  4:6');
+            assertEqual (iarr, [14:-2:5, 19, -3:-1:-6, 4:6])
+        end
+        
+        %--------------------------------------------------------------------------
+        % Enclosing brackets
+        %--------------------------------------------------------------------------
+        function test_array_scalar (~)
+            % Test constructor with no arguments
```

Comment:
oh yes it is

Reply:
I refer the honourable gentleman to my answer to a previous comment.